src/pydicom/pixels/decoders/base.py

```diff
@@ -1064,20 +1093,19 @@ def _test_for(self, test: str, index: int | None = None) -> bool:
                 and self.get_option("pixel_vr") == "OW"
             )
 
-        if test == "sign_correction":
-            use_j2k_correction = (
-                self.transfer_syntax in JPEG2000TransferSyntaxes
-                and self.photometric_interpretation in (PI.MONOCHROME1, PI.MONOCHROME2)
-                and self.get_option("apply_j2k_sign_correction", False)
+        if test == "j2k_corrections":
+            return self.transfer_syntax in JPEG2000TransferSyntaxes and (
+                self.get_option("apply_j2k_sign_correction", False)
```

Comment:
What about the check for monochrome photemetric interpretation for the sign correction case? Doesn't it still make sense?

Reply:
I don't think it's really necessary as even with it in we still potentially do a bit-shift in the last part:
```python
if runner._test_for("j2k_corrections"):  # If this test fails
    arr = _apply_j2k_corrections(arr, runner)
elif runner._test_for("jls_sign_correction"):
     arr = _apply_jls_sign_correction(arr, runner)
elif runner._test_for("shift_correction"):  # this may still pass
     arr = _correct_unused_bits(arr, runner)
```
But it's better to do the first test as it takes into account the J2K codestream precision for each frame + *Bits Stored* from the dataset, whereas the last test only takes into account *Bits Stored* from the dataset.